examples/advanced_operations/add_smart_display_ad.py

```diff
@@ -0,0 +1,267 @@
+def _create_responsive_display_ad(client, customer_id, ad_group_resource_name,
+                                  marketing_image_asset_resource_name,
+                                  square_marketing_image_asset_resource_name):
+    ad_group_ad_operation = client.get_type('AdGroupAdOperation', version='v2')
+    ad_group_ad = ad_group_ad_operation.create
+    ad_group_ad.ad_group.value = ad_group_resource_name
+    ad_group_ad.status = client.get_type(
+        'AdGroupAdStatusEnum', version='v2').PAUSED
+    ad = ad_group_ad.ad
+    final_url = ad.final_urls.add()
+    final_url.value = 'https://www.example.com'
+    responsive_display_ad = ad.responsive_display_ad
+    headline = responsive_display_ad.headlines.add()
+    headline.text.value = 'Travel'
+    responsive_display_ad.long_headline.text.value = 'Travel the World'
+    description = responsive_display_ad.descriptions.add()
+    description.text.value = 'Take to the air!'
+    responsive_display_ad.business_name.value = 'Google'
+    marketing_image = responsive_display_ad.marketing_images.add()
+    marketing_image.asset.value = marketing_image_asset_resource_name
+    square_marketing_image = responsive_display_ad.square_marketing_images.add()
+    square_marketing_image.asset.value = (
+        square_marketing_image_asset_resource_name)
+    responsive_display_ad.call_to_action_text.value = 'Shop Now'
+    responsive_display_ad.price_prefix.value = 'as low as'
+    responsive_display_ad.promo_text.value = 'Free shipping!'
+
+    ad_group_ad_service = client.get_service('AdGroupAdService', version='v2')
+
+    try:
+        ad_group_ad_response = ad_group_ad_service.mutate_ad_group_ads(
+            customer_id, [ad_group_ad_operation])
+    except google.ads.google_ads.errors.GoogleAdsException as ex:
+        print(f'Request with ID "{ex.request_id}" failed with status '
+              f'"{ex.error.code().name}" and includes the following errors:')
+        for error in ex.failure.errors:
+            print(f'\tError with message "{error.message}".')
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print(f'\t\tOn field: {field_path_element.field_name}')
+        sys.exit(1)
+
+    return ad_group_ad_response.results[0].resource_name
+
+
+if __name__ == '__main__':
+    # GoogleAdsClient will read the google-ads.yaml configuration file in the
+    # home directory if none is specified.
+    google_ads_client = (google.ads.google_ads.client.GoogleAdsClient
+                         .load_from_storage())
+
+    parser = argparse.ArgumentParser(
+        description=('Creates a Smart Display campaign, and an ad group that '
+                     'are then used to create a responsive display ad.'))
+    # The following argument(s) should be provided to run the example.
+    parser.add_argument('-c', '--customer_id', type=str,
+                        required=True, help='The Google Ads customer ID.')
```

Comment:
Golden has optional parameters for marketing_image_asset and square_marketing_image_asset

https://github.com/googleads/google-ads-php/blob/3dc0d995dd292a3cb77671917da6528011f7a3d2/examples/AdvancedOperations/AddSmartDisplayAd.php#L89

Reply:
Added.